Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM debian:stretch-slim
+
+ARG APACHE_SPARK_VERSION=2.2.0
+ARG HADOOP_VERSION=2.7
+ENV SPARK_NAME=spark-${APACHE_SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}
+
+ENV SPARK_DIR /opt/${SPARK_NAME}
+ENV SPARK_HOME /usr/local/spark
+
+RUN mkdir -p /usr/share/man/man1 && \
+    apt-get update && \
+    apt-get install -y --no-install-recommends \
+    openjdk-8-jre-headless  ca-certificates-java curl && \
+    rm -rf /var/lib/apt/* && \
+    curl http://d3kbcqa49mib13.cloudfront.net/${SPARK_NAME}.tgz | \
+    tar xzf - -C /opt && \
+    mkdir -p /root/.sbt/launchers/0.13.13 && \
+    curl -L -o /root/.sbt/launchers/0.13.13/sbt-launch.jar http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/sbt-launch/0.13.13/sbt-launch.jar && \
+    apt-get remove -y curl && \
+    apt-get clean
+
+RUN ln -s $SPARK_DIR $SPARK_HOME
```

Comment:
We're trying to maintain the Dockerfiles with one single `RUN` command.
Are you splitting this into three different ones for dev purposes? If so, it lgtm

Reply:
the purpose is docker layers. Because we don't use CI yet for publishing it allows caching of previous steps locally which speed-up building a lot. But even for CI it will speed up the process if we configure it properly.

Reply:
For dev purposes I noticed that speed-up and agree on that for sure. For prod purposes, you'll need to deal with Infra team (but I share your pov)

Reply:
we build images in travis no? infra doesn't manage travis.
by default travis doesn't use docker cache, but there is a workaround with `docker pull` http://atodorov.org/blog/2017/08/07/faster-travis-ci-tests-with-docker-cache/
not sure it worth to do in our case though, depends on the size of last layer, need to test.

Btw base image for engine also use multiple runs https://github.com/src-d/jupyter-spark-docker/blob/master/Dockerfile
So as engine itself https://github.com/src-d/engine/blob/master/Dockerfile